data/src/appearance/theme.rs

```diff
@@ -128,72 +128,252 @@ pub struct Buffer {
+    pub wallops: OptionalTextStyle,
+    #[serde(default)]
+    pub default: TextStyle,
 }
 
 #[derive(Debug, Clone, Copy, Serialize, Deserialize, Default)]
 pub struct Text {
-    #[serde(default = "default_transparent", with = "color_serde")]
-    pub primary: Color,
-    #[serde(default = "default_transparent", with = "color_serde")]
-    pub secondary: Color,
-    #[serde(default = "default_transparent", with = "color_serde")]
-    pub tertiary: Color,
-    #[serde(default = "default_transparent", with = "color_serde")]
-    pub success: Color,
-    #[serde(default = "default_transparent", with = "color_serde")]
-    pub error: Color,
+    #[serde(default)]
+    pub primary: TextStyle,
+    #[serde(default)]
+    pub secondary: TextStyle,
+    #[serde(default)]
+    pub tertiary: TextStyle,
+    #[serde(default)]
+    pub success: TextStyle,
+    #[serde(default)]
+    pub error: TextStyle,
 }
 
-impl Default for Colors {
+impl Default for Styles {
     fn default() -> Self {
         toml::from_str(DEFAULT_THEME_CONTENT).expect("parse default theme")
     }
 }
 
+#[derive(Clone, Copy, Debug)]
+pub struct TextStyle {
+    pub color: Color,
+    pub font_style: Option<FontStyle>,
+}
+
+impl Default for TextStyle {
+    fn default() -> Self {
+        Self {
+            color: Color::TRANSPARENT,
+            font_style: None,
+        }
+    }
+}
+
+impl<'de> Deserialize<'de> for TextStyle {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        #[derive(Deserialize)]
+        #[serde(untagged)]
+        enum Data {
+            Basic(String),
+            #[serde(rename_all = "kebab-case")]
+            Extended {
+                color: String,
+                font_style: Option<FontStyle>,
+            },
+        }
+
+        let data = Data::deserialize(deserializer)?;
+
+        let (hex, font_style) = match data {
+            Data::Basic(color) => (color, None),
+            Data::Extended { color, font_style } => (color, font_style),
+        };
+
+        Ok(TextStyle {
+            color: hex_to_color(&hex).unwrap_or(Color::TRANSPARENT),
+            font_style,
+        })
+    }
+}
+
+impl Serialize for TextStyle {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        #[derive(Serialize)]
+        #[serde(rename_all = "kebab-case")]
+        struct Data {
+            color: String,
+            font_style: Option<FontStyle>,
+        }
+
+        let hex = color_to_hex(self.color);
+
+        if self.font_style.is_some() {
+            Data {
+                color: hex,
+                font_style: self.font_style,
+            }
+            .serialize(serializer)
+        } else {
+            hex.serialize(serializer)
+        }
+    }
+}
+
+#[derive(Clone, Copy, Debug, Default)]
+pub struct OptionalTextStyle {
+    pub color: Option<Color>,
+    pub font_style: Option<FontStyle>,
+}
+
```

Comment:
Can we just make this version `TextStyle` and use it both where `TextStyle` is used and where `OptionalTextStyle` is used? If someone omits `color` where we previously made it required, we can just `unwrap_or(Transparent)` and clean this up a bunch. 

Or we can go the other direction, we always make `Color` non-optional but we use `default = transparent` w/ serde so if they don't define it, we set it as `transparent`, and does that have the same effect as if it wasn't defined?

Just trying to think of ways to not need both of these 😬